client/me/billing-history/transactions-header.jsx

```diff
@@ -89,32 +98,32 @@ class TransactionsHeader extends React.Component {
 		);
 	}
 
-	setFilter( filter ) {
-		this.setState( { activePopover: '' } );
-		this.props.onNewFilter( filter );
-	}
-
 	renderDatePopover() {
-		const previousMonths = range( 6 ).map( function( n ) {
-			return this.props.moment().subtract( n, 'months' );
-		}, this );
+		const { filter, dateFilters } = this.props;
+		const selectedFilter = find( dateFilters, filterMeta =>
+			isEqual( filterMeta.value, filter.date )
```

Comment:
Lodash's `find` has an object predicate shorthand which could help shorten this and avoid `isEqual` usage. So

```
const selectedFilter = find( dateFilters, filterMeta =>
	isEqual( filterMeta.value, filter.date )
);
```

would become:

```
const selectedFilter = find( dateFilters, { value: filter.date } );
```